op-bindings/etherscan/client.go

```diff
@@ -0,0 +1,196 @@
+package etherscan
+
+import (
+	"context"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"io"
+	"net/http"
+	"strings"
+	"time"
+
+	"github.com/ethereum-optimism/optimism/op-service/retry"
+)
+
+type client struct {
+	baseUrl string
+}
+
+type apiResponse struct {
+	Status  string      `json:"status"`
+	Message string      `json:"message"`
+	Result  interface{} `json:"result"`
+}
+
+type rpcResponse struct {
+	JsonRpc string      `json:"jsonrpc"`
+	Id      int         `json:"id"`
+	Result  interface{} `json:"result"`
+}
+
+type TxInfo struct {
+	To    string `json:"to"`
+	Input string `json:"input"`
+}
+
+const apiMaxRetries = 3
+const apiRetryDelay = time.Duration(2) * time.Second
+const errRateLimited = "Max rate limit reached"
+
+func NewClient(baseUrl, apiKey string) *client {
```

Comment:
Can this be private now that there's `NewEthereumClient` and `NewOptimismClient`?

Reply:
sounds good!